app/services/follow_service.rb

```diff
@@ -71,7 +71,7 @@ def request_follow!
     if @target_account.local?
       LocalNotificationWorker.perform_async(@target_account.id, follow_request.id, follow_request.class.name, 'follow_request')
     elsif @target_account.activitypub?
-      ActivityPub::DeliveryWorker.perform_async(build_json(follow_request), @source_account.id, @target_account.inbox_url)
+      ActivityPub::DeliveryWorker.perform_async(build_json(follow_request), @source_account.id, @target_account.inbox_url, { 'destroy_on_failure' => { 'class' => 'FollowRequest', 'id' => follow_request.id.to_s } })
```

Comment:
I wonder if this could be a boolean to destroy or not, and then we could pull out the class/id/etc from the passed in object ... or if that's too flaky b/c we're sending in json and not the record...